File: specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/examples-js/HcxEnterpriseSites_CreateOrUpdate.js
const { AzureVMwareSolutionAPI } = require("@azure/arm-avs");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Create or update an activation key for on-premises HCX site
 *
 * @summary Create or update an activation key for on-premises HCX site
 * x-ms-original-file: specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/examples/HcxEnterpriseSites_CreateOrUpdate.json
 */
async function hcxEnterpriseSitesCreateOrUpdate() {
  const subscriptionId =
    process.env["AVS_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["AVS_RESOURCE_GROUP"] || "group1";
  const privateCloudName = "cloud1";
  const hcxEnterpriseSiteName = "site1";
  const hcxEnterpriseSite = {};
  const credential = new DefaultAzureCredential();
  const client = new AzureVMwareSolutionAPI(credential, subscriptionId);
  const result = await client.hcxEnterpriseSites.createOrUpdate(
    resourceGroupName,
    privateCloudName,
    hcxEnterpriseSiteName,
    hcxEnterpriseSite
  );
  console.log(result);
}

File: specification/vmware/resource-manager/Microsoft.AVS/AVS/stable/2025-09-01/examples-java/HcxEnterpriseSites_CreateOrUpdate.java

/**
 * Samples for HcxEnterpriseSites CreateOrUpdate.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-09-01/HcxEnterpriseSites_CreateOrUpdate.json
     */
    /**
     * Sample code: HcxEnterpriseSites_CreateOrUpdate.
     * 
     * @param manager Entry point to AvsManager.
     */
    public static void hcxEnterpriseSitesCreateOrUpdate(com.azure.resourcemanager.avs.AvsManager manager) {
        manager.hcxEnterpriseSites().define("site1").withExistingPrivateCloud("group1", "cloud1").create();
    }
}
